### PR TITLE
Rebalance testcases into groups

### LINKED ISSUE #10235: G11 is too heavy in horizontal comparison in the test groups
Ref: https://github.com/pingcap/tiflow/blob/89e57d7a6e5d3114bcac900ecd3c0a905b849ea8/dm/tests/run_group.sh#L28

Usually the [pull_dm_integration_test] CI job takes 50 minutes, prepare stage costed 10 minutes. `G11` test group took more than 40 minutes and became the critical path.

image
Example build url: https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftiflow%2Fpull_dm_integration_test/detail/pull_dm_integration_test/2713/pipeline

### PR DESCRIPTION
**Pull request by Benjamin2037**
### What problem does this PR solve?

Issue Number: close #10235

### What is changed and how it works?
Rebance testcases in test group to reduce total testing time.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: dm/tests/run_group.sh [hunk 1]
```diff
@@ -25,7 +25,14 @@ groups=(
 	["G09"]="import_v10x sharding2 ha"
 	["G10"]="start_task print_status http_apis new_relay all_mode"
 	# `others others_2 others_3` tests of old pipeline
-	["G11"]="adjust_gtid async_checkpoint_flush binlog_parse case_sensitive checkpoint_transaction check_task dm_syncer downstream_diff_index downstream_more_column drop_column_with_index duplicate_event expression_filter extend_column fake_rotate_event foreign_key full_mode gbk gtid ha_cases http_proxies lightning_load_task lightning_mode metrics new_collation_off only_dml openapi s3_dumpling_lightning sequence_sharding_optimistic sequence_sharding_removemeta shardddl_optimistic slow_relay_writer sql_mode sync_collation tracker_ignored_ddl validator_basic validator_checkpoint incompatible_ddl_changes"
+	["G11"]="adjust_gtid async_checkpoint_flush binlog_parse case_sensitive”
```
Comment (lichunzhu) [inline dm/tests/run_group.sh:28]:
```suggestion
	["G11"]="adjust_gtid async_checkpoint_flush binlog_parse case_sensitive"
```

**Comment by purelind** [comment]
Observing the existing build history, it is found that except for group11, the execution time of other group tests is generally around 4 minutes. It is suggested to consider dispersing the tests in group11 among the existing groups instead of adding more groups. 
image

**Review by lichunzhu — APPROVED** [review]

**Review by D3Hunter — COMMENTED** [review]
i remember `ha_casesXXX` has many cases inside, not sure this could speed up or not

**Review by lance6716 — APPROVED** [review]

**Pull request merged**